src/Lean/Elab/Tactic/BuiltinTactic.lean

```diff
@@ -323,6 +323,9 @@ def forEachVar (hs : Array Syntax) (tac : MVarId → FVarId → MetaM MVarId) :
 @[builtin_tactic Lean.Parser.Tactic.substVars] def evalSubstVars : Tactic := fun _ =>
   liftMetaTactic fun mvarId => return [← substVars mvarId]
 
+/-- `subst_eqs` applies `subst` to all equalities in the context as long as it makes progress. -/
```

Comment:
I think this formulation is easier to understand:
```suggestion
/-- Repeatedly substitutes according to the equality proof hypotheses in the context, replacing the left side of the equality with the right, until no more progress can be made. -/
```

But what happens if there are no such hypotheses? Does it fail or is it a no-op?

And from what I can see, the implementation doesn't use `subst`, but rather `casesOn`. But that's a bit beside the point - tactic users shouldn't have to care which term they get, at least not for things in `Prop`.